common/djangoapps/entitlements/models.py

```diff
@@ -307,26 +311,33 @@ def get_active_entitlements_for_user(cls, user):
         ).select_related('user').select_related('enrollment_course_run')
 
     @classmethod
-    def does_user_have_active_entitlements(cls, user):
+    def get_fullfillable_entitlements(cls, user):
         """
-        Checks whether or not a user has any active entitlements
+        Returns the most recent Course Entitlement for a Course that a user can use
+
+        Arguments:
+            user (User): The user we are looking at the entitlements of.
+            course_uuid: The Course UUID of the course we are looking for.
+
+        Returns
+            list<CourseEntitlements>: A list of course Entitlements that a user can enroll in. These must not be expired
+                and not have a course run already assigned to it.
         """
-        count = cls.objects.count(
-            user=user
-        ).exclude(
-            expired_at__isnull=False,
-            enrollment_course_run=None
-        ).select_related('user').select_related('enrollment_course_run')
-        return count > 0
+
+        try:
+            return cls.objects.filter(
+                user=user,
+            ).exclude(
+                expired_at__isnull=False,
+                enrollment_course_run__isnull=False
+            ).order_by('created')
+        except CourseEntitlement.DoesNotExist:
+            return []
 
     @classmethod
     def get_fullfillable_entitlement_for_user(cls, user, course_run_key):
```

Comment:
this method name could be clearer, preferably indicating that the entitlement is being found not only for a given user but also for a given course run